azext_iot/central/providers/destination_provider.py

```diff
@@ -57,7 +56,7 @@ def add_destination(
         AdxDestinationV1_1_preview,
     ]:
         if destination_id in self._destinations:
-            raise CLIError("Destination already exists")
+            raise BadRequestError("Destination already exists")
```

Comment:
Hmm in this case the request to the service is well formed, but the user destination Id already exists...a '409' like error seems more appropriate compared to a 400 like error.

Reply:
There were a few of these where I wasn't sure. There isn't really a `409 Conflict` type of azclierror, but I think you're right in that `ClientRequestError` seems a bit more apt here than `BadRequestError` - because the request is formed correctly, there's just an error handling it.

I'll update any of the `Resource already exists` type of errors to raise a `ClientRequestError` instead

Reply:
Or rather (now that I'm reading this again), should we only update the situations where we're hitting an internal cache to use a `ClientRequestError` - and leave other instances (`device create`, `policy create`, etc.) as a `BadRequestError` if the resources already exist?

Reply:
Yea the internal cache cases where the client does not send the request are apt for `ClientRequestError`. This occurs elsewhere, in say ....`device create`

![image](https://user-images.githubusercontent.com/28658112/156458068-8960d159-4ce3-433c-a683-2a598c174687.png)